020-Getting-started/110-nuxt.mdx

```diff
@@ -0,0 +1,597 @@
+    <NuxtPage />
+  </main>
+</template>
+```
+
+Ensure the `<NuxtPage />` component is before the closing `</main>` element.
+
+## Query and list the posts
+
+Now the Nuxt app is ready to for Xata to be integrated into the codebase.
+
+Let's start by making use of
+[Nuxt's server routes](https://nuxt.com/docs/guide/directory-structure/server#server-routes) to retrieve all the blog posts from Xata.
+Create `server/api/posts.ts` with the following contents:
+
+```ts title="server/api/posts.ts"
+import { getXataClient } from '../../src/xata';
+const xata = getXataClient();
+
+export default defineEventHandler(async (event) => {
+  const posts = await xata.db.Posts.getAll();
+
+  return posts;
+});
+```
+
+First, import the `getXataClient` function and assign the result of calling that function to a variable named `xata`. This client uses the configuration set in the `.env` file.
+
+Then, define and export a call to the `defineEventHandler` that returns a handler function. Within that handler function use the `xata` client instance to get all
+the posts stored in the database. This is achieved via the auto-generated `Posts` property which exposes a number of
+helper functions. In this case, use the [`getAll`](https://xata.io/docs/typescript-client/get#the-typescript-sdk-functions-for-querying)
+function to get all the Post records. Finally, return the `posts` as the function return value.
+
+You can access `http://localhost:3000/api/posts` to see the posts represented as JSON.
+
+<Alert status="warning">
+  `getAll()` returns all the records in the query results. This is dangerous on large tables (more than 10,000 records),
+  as it will potentially load a lot of data into memory and create a lot of requests to the server. In most situations,
+  you should use `getMany()` or `getPaginated()`. See the [querying
+  documentation](https://xata.io/docs/typescript-client/get#the-typescript-sdk-functions-for-querying) for more
+  information.
+</Alert>
+
+Next, create a `pages/index.vue` file as a landing page to load the data and render the list of blog posts:
+
+```vue title="pages/index.vue"
+<script setup lang="ts">
+const { data: posts } = await useFetch(`/api/posts`);
+</script>
+
+<template>
+  <div class="w-full max-w-5xl mt-16">
+    <p v-if="posts && posts.length === 0">No blog posts found</p>
+    <div v-for="post in posts" class="mb-16">
+      <p class="text-xs mb-2 text-purple-950 dark:text-purple-200">
+        {{ new Date(post.pubDate || '').toDateString() }}
+      </p>
+      <h2 class="text-2xl mb-2">
+        <a :href="`/posts/${post.slug}`">{{ post.title }}</a>
+      </h2>
+      <p class="text-purple-950 dark:text-purple-200 mb-5">{{ post.description }}</p>
+      <a
+        :href="`/posts/${post.slug}`"
+        class="px-4 py-2 font-semibold text-sm bg-purple-700 text-white rounded-lg shadow-sm w-fit"
+      >
+        Read more &rarr;
+      </a>
+    </div>
+  </div>
+</template>
+```
+
+Let's walk through the code above.
```

Comment:
This only rendered after restarting the runserver.

Reply:
Added note